runtimes/runtimes/operational-telemetry/operational-event-validator.test.ts

```diff
@@ -0,0 +1,59 @@
+import { OperationalEventValidator } from './operational-event-validator'
+import assert from 'assert'
+
+describe('OperationalEventValidator', () => {
+    let validator: OperationalEventValidator
+
+    before(() => {
+        validator = new OperationalEventValidator()
+    })
+
+    describe('validateEvent', () => {
+        it('should return true for a valid CaughtErrorEvent', () => {
+            const event = {
+                name: 'CaughtErrorEvent',
+                timestamp: 100000,
+                errorType: 'Error',
+            }
+
+            const result = validator.validateEvent(event)
+            assert.ok(result)
+        })
+
+        it('should return true for a valid ResourceUsageMetric', () => {
+            const event = {
+                name: 'ResourceUsageMetric',
+                timestamp: 100000,
+                userCpuUsage: 100,
+                systemCpuUsage: 1,
+                heapUsed: 2,
+                heapTotal: 3,
+                rss: 4,
+            }
+
+            const result = validator.validateEvent(event)
+            assert.ok(result)
+        })
+
+        it('should return true for a valid ServerCrashEvent', () => {
+            const event = {
+                name: 'ServerCrashEvent',
+                timestamp: 100000,
+                crashType: 'Error',
+            }
+
+            const result = validator.validateEvent(event)
+            assert.ok(result)
+        })
+
+        it('should return false when missing fields', () => {
```

Comment:
What's the strategy for fields not in schema? Should we remove them or fail validation?

Reply:
I think we should fail validation, there is a special configuration in the json schema for that, namely `unevaluatedProperties: false`. Unfortunately it's not available yet in http://json-schema.org/draft-07/schema# that we use, and I couldn't get the `ajv` library work with the newer schema. We can improve it in the future.